router/router.go

```diff
@@ -0,0 +1,139 @@
+package router
+
+import (
+	"net/http"
+
+	"github.com/go-chi/chi"
+	"github.com/netlify/netlify-commons/tracing"
+	"github.com/rs/cors"
+	"github.com/sebest/xff"
+	"github.com/sirupsen/logrus"
+)
+
+type chiWrapper struct {
+	chi chi.Router
+}
+
+// Options configures specifics of the default router
+type Options struct {
```

Comment:
I'm more in favor of functional options as a configuration mechanism: 
* https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis
* https://commandcenter.blogspot.com/2014/01/self-referential-functions-and-design.html

In a first look, it could seem weird but it facilitates final users to compose semantically the options to pass, becoming more scalable from the point of view of the code as easy to test and distribute. 

Some big Golang public libs use this pattern:
* New preview version of AWS sdk: https://github.com/aws/aws-sdk-go-v2/blob/a3571312476e84b6837ca42f6c8a4654633d0a53/aws/request.go#L142
*https://github.com/kubernetes/kubernetes/blob/a33840e0232aaf2b5c14c1c754d9189d7e841ec1/pkg/kubelet/container/runtime.go#L611
* https://github.com/uber/kraken/blob/128c7e2e4c264c780edc2d764c6ec12484e1dac8/lib/backend/gcsbackend/client.go#L75
* https://github.com/gorilla/csrf/blob/master/options.go#L8